streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -519,7 +519,7 @@ public void shouldBeAbleToQueryFilterState() throws Exception {
         kafkaStreams = new KafkaStreams(builder.build(), streamsConfiguration);
         kafkaStreams.start();
 
-        waitUntilAtLeastNumRecordProcessed(outputTopic, 2);
+        waitUntilAtLeastNumRecordProcessed(outputTopic, 1);
```

Comment:
This is the effect of the optimization: we will only sent down one record now, and the second `null` record will be omitted.